KTL.js

```diff
@@ -7362,8 +7364,10 @@ function Ktl($, appInfo) {
             })
 
             function startQtScanning() {
-                ktl.core.infoPopup();
-                showProgress();
+                if (!quickToggleParams.hideNotification) {
```

Comment:
Just a suggestion. To avoid a double negative ( ! -> hide), you could have use "showNotification". Readability would be simpler, if (params.show) then show() instead of if( not params.hide) then show().

Reply:
Hi Charles
I agree, that was my original choice but due to these being new variables and users already using existing parameters I would have had to check if the variables existed then decide if they were true.  This may be something we could change in the future giving the users enough notice.  My original code was

`if (quickToggleParams.hideNotification || quickToggleParams.hideNotification === 'true'`

But to make the code simpler I chose to do the double negative as this would have less impact on users.

If I missed a trick, I'd be more than happy to update the code.
Craig

Reply:
Hi Charles
The issue I run into is, if users who already have the quickToggleParams but don't have the new parameters in your example 
showNotification & showSpinner 
Will from now on always have their spinner and notification hidden.  I was aiming for some continuation until we could allow everyone time to update their params.  If there is a way to do this that I am missing, then I am more than happy to change my code.

Craig

Reply:
Hi Craig
These new parameters' default values could be `true`. This should solve the issue you mention. 

I'm aware that the configuration is wrongly done to support a partial config object.
This line
`cfgObj.quickToggleParams && (quickToggleParams = cfgObj.quickToggleParams);`
should become
`cfgObj.quickToggleParams && (quickToggleParams = {...quickToggleParams, ...cfgObj.quickToggleParams});`

What this does is the spread operator '...' enumerates properties of what is already there (the defaults, aka quickToggleParams) with what's coming in (user's parameters, aka cfgObj.quickToggleParams) all in a new object. Same-name Parameters overwrite the previous ones in order of enumeration.  However, in our case, it does not overwrite the defaults with the undefineds in the user's. Default values set to `true` will remain `true` if they are not set by the users.

Charles

Reply:
Ok thank you I'll get that sorted.

Craig